### PR TITLE
DOCKERIZE

### PR DESCRIPTION
**Pull request by JAremko**
Use Docker and  Xpra to isolate LalkaChat and its dependencies. Allow browser based configuration.

### CODE REVIEW

**Comment by DeForce** [comment]
Please format this pull request in a more "standard" way.

File: modules/messaging/webchat.py [hunk 1]
```diff
@@ -205,7 +205,7 @@ def __init__(self, conf_folder, **kwargs):
             'id': DEFAULT_PRIORITY
         }
         conf_dict['server'] = OrderedDict()
-        conf_dict['server']['host'] = '127.0.0.1'
+        conf_dict['server']['host'] = '0.0.0.0'
```
Comment (DeForce) [inline modules/messaging/webchat.py]: No, alter it yourself after docker container initialization

File: gui.py [hunk 2]
```diff
@@ -778,10 +778,13 @@ def __init__(self, parent, title, url, **kwargs):
...
         self.toolbar = MainMenuToolBar(self, main_class=self)
-        self.browser_window = chromectrl.ChromeCtrl(self, useTimer=False, url=str(url), hasNavBar=False)
+
+        browser_window  = chromectrl.ChromeCtrl(self, useTimer=False, url=str(url), hasNavBar=False)
+        if hasattr(browser_window, 'stub') != True:
+            self.browser_window = chromectrl.ChromeCtrl(self, useTimer=False, url=str(url), hasNavBar=False)
+            vbox.Add(self.browser_window, 1, wx.EXPAND)
 
         vbox.Add(self.toolbar, 0, wx.EXPAND)
-        vbox.Add(self.browser_window, 1, wx.EXPAND)
```
Comment (DeForce) [inline gui.py]: browser window should be added after toolbar

File: gui.py [hunk 3]
```diff
@@ -778,10 +778,13 @@ def __init__(self, parent, title, url, **kwargs):
         # Creating main gui window
         vbox = wx.BoxSizer(wx.VERTICAL)
         self.toolbar = MainMenuToolBar(self, main_class=self)
-        self.browser_window = chromectrl.ChromeCtrl(self, useTimer=False, url=str(url), hasNavBar=False)
+
+        browser_window  = chromectrl.ChromeCtrl(self, useTimer=False, url=str(url), hasNavBar=False)
+        if hasattr(browser_window, 'stub') != True:
```
Comment (DeForce) [inline gui.py]: It is all around wrong, you are creating 2 chrome windows, and will be an exception if it is stub.

File: docker/stubs/cefpython3/wx/chromectrl.py [hunk 4]
```diff
@@ -0,0 +1,9 @@
+class ChromeCtrl:
```
Comment (DeForce) [inline docker/stubs/cefpython3/wx/chromectrl.py]: Stubs shouldn't exist

File: docker/xpra-www/include/aurora/aurora-xpra.js [hunk 5]
```diff
@@ -0,0 +1,48 @@
+// based on aurora-websocket.js https://github.com/fsbdev/aurora-websocket
```
Comment (DeForce) [inline docker/xpra-www/include/aurora/aurora-xpra.js]: js files shouldn't exist, but downloaded from somewhere

**Comment by JAremko** [comment]
czt, uzbagoisa :holypeka:

**Comment by wdcct** [comment]
Что я сейчас прочитал? WTF?

**Comment by Hukutoss** [comment]
Считаю это лучшим коммитом за всё время :peka:

**Comment by robocobra** [comment]
@JAremko  Will it work on Windows? I can't quite understand Your PR message

**Comment by JAremko** [comment]
@robocobra  
>  I can't quite understand Your PR message  

Yeah.. the PR message.. Well... It's kind "insider joke"

>  Will it work on Windows?   

Sure it will! You can try my demo build  https://hub.docker.com/r/jare/lalkachat/ :wink:

**Comment by robocobra** [comment]
@JAremko  Thx! I'll try it.

**Comment by robocobra** [comment]
@JAremko  WOW! It actually works!  But I found a bug:  



The left box seems to be too small.  Other than that it is AWESOME!

Solves #169

**Comment by JAremko** [comment]
> The left box seems to be too small.    

@robocobra  Don't look at me, look at @DeForce  :wink:

**Comment by DeForce** [comment]
Created and solved #170

**Comment by robocobra** [comment]
Ну вот, а говорил что незамаржеш :pled:

**Comment by DeForce** [comment]
Solves #152

**Pull request merged**